packages/core/src/config/types.ts

```diff
@@ -34,6 +34,7 @@ export interface Config<P = APIEvent> {
     onSessionChange?: (oldSession: MetaSession | null, newSession: MetaSession) => void;
     samplingRate?: number;
     sampler?: (context: SamplingContext) => number;
+    sessionIdGenerator?: () => string;
```

Comment:
Nitpicky: I would expect a function to be named as an imperative verb (i.e. `generateSessionId()`) instead of a noun where it might indicate that it is an object. Feel free to merge though 🪨

Reply:
Argh sorry merged before i read the comment. 
Will rename it!